internal/website/content/howto/runtimervar/_index.md

```diff
@@ -0,0 +1,108 @@
+
+## Constructors versus URL openers
+
+The easiest way to open a variable is using [`runtimevar.OpenVariable`][] and a
+URL pointing to the variable. See [Concepts: URLs][] for more
+details. If you need fine-grained control over the connection settings, you can
+call the constructor function in the driver package directly (like
+`awsparamstore.OpenVariable`). This guide shows how to use both forms for
+each storage provider.
+
+[Concepts: URLs]: {{< ref "/concepts/urls.md" >}}
+
+[`runtimevar.OpenVariable`]:
+https://godoc.org/gocloud.dev/runtimevar#OpenVariable
+
+## GCP Runtime Configurator {#gcp-url}
+
+To open a variable stored in [GCP Runtime Configurator][] via a URL, you can use
+the `runtimevar.OpenVariable` function as follows. It uses the `string`
+[decoder][].
+
+{{< goexample
+"gocloud.dev/runtimevar/gcpruntimeconfig.Example_openVariableFromURL" >}}
+
+[decoder]: https://godoc.org/gocloud.dev/runtimevar#Decoder
+[GCP Runtime Configurator]: https://cloud.google.com/deployment-manager/runtime-configurator/
+
+## GCP Runtime Configurator Constructor {#rc-ctor}
+
+The [`gcpruntimeconfig.OpenVariable`][] constructor opens a Runtime Configurator
+variable.
+
+{{< goexample
+"gocloud.dev/runtimevar/gcpruntimeconfig.Example_openVariableHowto" >}}
+
+[`gcpruntimeconfig.OpenVariable`]: https://godoc.org/gocloud.dev/runtimevar/gcpruntimeconfig#OpenVariable
+
+## AWS Parameter Store {#ps-url}
+
+To open a variable stored in [AWS Parameter Store][] via a URL, you can use the
+`runtimevar.OpenVariable` function as follows. It uses the `string` [decoder][].
+
+{{< goexample
+"gocloud.dev/runtimevar/awsparamstore.Example_openVariableFromURL" >}}
+
+[AWS Parameter Store]:
+https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html
+
+## AWS Parameter Store Constructor {#ps-ctor}
+
+The [`awsparamstore.OpenVariable`][] constructor opens a Parameter Store
+variable.
+
+{{< goexample "gocloud.dev/runtimevar/awsparamstore.Example_openVariableHowto" >}}
+
+[`awsparamstore.OpenVariable`]:
+https://godoc.org/gocloud.dev/runtimevar/awsparamstore#OpenVariable
+
+## etcd {#etcd-url}
+
+The Go CDK supports using [etcd][] for storing variables locally or
+remotely. To open a variable stored in `etcd` via a URL, you can use the
+`runtimevar.OpenVariable` function as follows.
+
+{{< goexample
+"gocloud.dev/runtimevar/etcdvar.Example_openVariableFromURL" >}}
+
+[etcd]: https://etcd.io/
+
+## etcd Constructor
+
+The [`etcdvar.OpenVariable`][] constructor opens an etcd variable.
+
+[`etcdvar.OpenVariable`]:
+https://godoc.org/gocloud.dev/runtimevar/etcdvar#OpenVariable
+
+{{< goexample "gocloud.dev/runtimevar/etcdvar.Example_openVariableHowto" >}}
+
+## Fetching the latest value {#latest}
+
+Once we have an open variable, we can use the [`Variable.Latest`][] method to
+fetch its latest value. This method returns the latest good [`Snapshot`][] of
+the variable value, blocking if no good value has ever been received. To avoid
```

Comment:
I would emphasize this even more, that it only blocks if no good value has *ever* been received. In particular, it will not block if there's a good value and then there's a bad value (it will keep returning the good value).

Users might want to call `Snapshot` once during startup, or have `healthz` only succeed, based on getting a first good value. In fact, `Variable` implements `health.Checker` to make that easy if they happen to be using `server`.

Reply:
I added emphasis to "ever", I think the rest of the first paragraph is implied already. I suggest we leave the server/health reference out for now - until we have a howto for server. Then we can see where's the best place to describe this